diag_manager/fms_diag_axis_object.F90

```diff
@@ -543,6 +686,243 @@ subroutine get_domain_and_domain_type(diag_axis, axis_id, domain_type, domain, v
+        !! If it is get the starting and ending indices for that PE
+        call parent_axis%get_indices(compute_idx, subRegion%corners(:,i), starting_index, ending_index, &
+          need_to_define_axis)
+
+        !< If the PE's compute is not inside the subRegion, define a null subaxis and go to the next axis
+        if (.not. need_to_define_axis) then
+          call define_new_axis(diag_axis, parent_axis, naxis, axis_ids(i), &
+            subRegion, diag_null, diag_null)
+          cycle
+        endif
+
+        !< If it made it to this point, the current PE is in the subRegion!
+        write_on_this_pe = .true.
+
+        call define_new_axis(diag_axis, parent_axis, naxis, axis_ids(i), &
+            subRegion, starting_index, ending_index)
+        end select
+    enddo
+
+  end subroutine define_subaxis_index
+
+  !> @brief Fill in the subaxis object for a subRegion defined by lat lon
+  subroutine define_subaxis_latlon(diag_axis, axis_ids, naxis, subRegion, is_cube_sphere, write_on_this_pe)
+    class(fmsDiagAxisContainer_type), target, intent(inout) :: diag_axis(:)     !< Diag_axis object
+    integer,                                  INTENT(in)    :: axis_ids(:)      !< Array of axes_ids
+    integer,                                  intent(inout) :: naxis            !< Number of axis registered
+    type(subRegion_type),                     intent(in)    :: subRegion        !< SubRegion definition from the yaml
+    logical,                                  intent(in)    :: is_cube_sphere   !< .true. if this is a cubesphere
+    logical,                                  intent(out)   :: write_on_this_pe !< .true. if the subregion
+                                                                                !! is on this PE
+
+    real    :: lat(2)              !< Starting and ending lattiude of the subRegion
+    real    :: lon(2)              !< Starting and ending longitude or the subRegion
+    integer :: lat_indices(2)      !< Starting and ending latitude indices of the subRegion
+    integer :: lon_indices(2)      !< Starting and ending longitude indices of the subRegion
+    integer :: compute_idx(2)      !< Compute domain of the current axis
+    integer :: starting_index      !< Starting index of the subRegion for the current PE
+    integer :: ending_index        !< Ending index of the subRegion for the current PE
+    logical :: need_to_define_axis !< .true. if it is needed to define the subaxis
+    integer :: i                   !< For do loops
+
+    !< Get the rectangular coordinates of the subRegion
+    !! If the subRegion is not rectangular, the points outside of the subRegion will be masked
+    !! out later
+    select type (corners => subRegion%corners)
+    type is (real(kind=r4_kind))
+      lon(1) = minval(corners(:,1))
+      lon(2) = maxval(corners(:,1))
+      lat(1) = minval(corners(:,2))
+      lat(2) = maxval(corners(:,2))
+    end select
+
+    if (is_cube_sphere) then
```

Comment:
Can you label the `if`s and `do`s in this routine?  There's a lot of them and they are large.